File: Dummy/hawkshaw/library/datalayer/room/telephony/CallBlockEntity.java
package com.hawkshaw.library.datalayer.room.telephony;

import Y1.K;

public final class CallBlockEntity {
    private final boolean blockOutgoing;
    private int id;
    private final String name;
    private final String number;

    public CallBlockEntity(String str, String str2, boolean z4) {
        K.n(str, "name");
        K.n(str2, "number");
        this.name = str;
        this.number = str2;
        this.blockOutgoing = z4;
    }

    public final boolean getBlockOutgoing() {
        return this.blockOutgoing;
    }

    public final int getId() {
        return this.id;
    }

    public final String getName() {
        return this.name;
    }

    public final String getNumber() {
        return this.number;
    }

    public final void setId(int i5) {
        this.id = i5;
    }
}

File: app/src/main/java/com/hawkshaw/library/datalayer/room/telephony/CallBlockEntity.kt
package com.hawkshaw.library.datalayer.room.telephony

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Entity representing a phone number to be blocked
 */
@Entity(tableName = "call_block_entity")
data class CallBlockEntity(
    @ColumnInfo(name = "name")
    var name: String = "",
    
    @ColumnInfo(name = "number")
    var number: String = "",
    
    @ColumnInfo(name = "block_outgoing")
    var blockOutgoing: Boolean = false,

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id: Int = 0
) 